components/hab/src/command/pkg/uninstall.rs

```diff
@@ -162,6 +181,15 @@ fn maybe_delete(
         return Ok(false);
     }
 
+    let is_running = services.iter().any(|i| i.satisfies(ident));
+    if is_running {
+        ui.status(
+            Status::Skipping,
+            format!("{}. It is currently running in the supervisor", &ident),
+        )?;
```

Comment:
A lot of the output around services feels a bit misleading, at the code level, given that we're not actually determining that they're running here.

It feels like the `services` list is a special variant or subset of the `excludes` list, since they're also excluded identifiers... seems like the main difference is how we log when we come across one of them. Is there another difference that is perhaps implicit in the code that I'm not seeing (e.g., the `services` idents are fully-qualified, but the `excludes` can be fuzzy)?

Along those lines, if the service idents _can_ be fuzzy, this would seem to keep any satisfying package, rather than the specific one that is running (or I guess the "latest" package, which should work out to be the same thing).

If service idents are meant to always be fully-qualified, we could tighten this up by verifying that's the case in this code.

It might be worth adjusting the documentation comment to reflect that these service idents are just being treated as a special kind of `excludes`, and that no actual determination of "live-ness" or "service-ness" is being made here.

(Paging @fnichol on this, too, since he was doing some exploratory work a while back on expressing fully-qualified vs. fuzzy identifiers at the type level, which could come in handy here.)

Reply:
Yeah, this is a little bit about trying to separate concerns between `main.rs` and `uninstall.rs`.  I actually check for running services in `main.rs` as that is where all the sup API code is already.  But I want to output these messages *after* the `"Uninstalling {}"` line. This is the only way I had to do that.

I had an alternative design where `services` was a `Result<Option<Vec<PackageIdent>>>` but as @baumanj pointed out, this code is much cleaner.

Reply:
Yes, services are fully qualified, and we only exclude the running version.  This is so we can use uninstall to clean up old versions of running services. It doesn't have to be the latest even, just the version you've decided to run right now.

I'll update the doc comment to note that the running services are passed in.  My general assumption was that you shouldn't treat the commands as something run differently than the way they're run in `main.rs` right now.